libmamba/src/core/subdirdata.cpp

```diff
@@ -115,182 +166,226 @@ namespace mamba
         return last_write_time_valid;
     }
 
-    tl::expected<subdir_metadata, mamba_error> subdir_metadata::from_stream(std::istream& in)
+    const std::string& MSubdirMetadata::url() const
+    {
+        return m_http.url;
+    }
+
+    const std::string& MSubdirMetadata::etag() const
+    {
+        return m_http.etag;
+    }
+
+    const std::string& MSubdirMetadata::last_modified() const
     {
-        nlohmann::json j = nlohmann::json::parse(in);
-        subdir_metadata m;
+        return m_http.last_modified;
+    }
+
+    const std::string& MSubdirMetadata::cache_control() const
+    {
+        return m_http.cache_control;
+    }
+
+    bool MSubdirMetadata::has_zst() const
+    {
+        return m_has_zst.has_value() && m_has_zst.value().value && !m_has_zst.value().has_expired();
+    }
+
+    void MSubdirMetadata::store_http_metadata(http_metadata data)
+    {
+        m_http = std::move(data);
+    }
+
+    void MSubdirMetadata::store_file_metadata(const fs::u8path& file)
+    {
+#ifndef _WIN32
+        m_stored_mtime = fs::last_write_time(file);
+#else
+        // convert windows filetime to unix timestamp
+        m_stored_mtime = filetime_to_unix(fs::last_write_time(file));
+#endif
+        m_stored_file_size = fs::file_size(file);
```

Comment:
Seems like timestamp could use their own utility abstraction.

Reply:
Or we could remove the `filetime_to_unix` call, and always use `last_write_time` only.